rmf_site_editor/src/widgets/inspector/inspect_model_description/inspect_circle_collision.rs

```diff
@@ -0,0 +1,160 @@
+            Grid::new("inspect_circle_collision")
+                .num_columns(3)
+                .show(ui, |ui| {
+                    ui.label("Collision Radius");
+                    ui.add(
+                        DragValue::new(&mut new_circle_collision.radius)
+                            .clamp_range(0_f32..=std::f32::INFINITY)
+                            .speed(0.01),
+                    );
+                    ui.label("m");
+                    ui.end_row();
+
+                    ui.label("Offset");
+                    ui.label("x");
+                    ui.label("y");
+                    ui.end_row();
+
+                    ui.label("");
+                    ui.add(
+                        DragValue::new(&mut new_circle_collision.offset[0])
+                            .clamp_range(std::f32::NEG_INFINITY..=std::f32::INFINITY)
+                            .speed(0.01),
+                    );
+                    ui.add(
+                        DragValue::new(&mut new_circle_collision.offset[1])
+                            .clamp_range(std::f32::NEG_INFINITY..=std::f32::INFINITY)
+                            .speed(0.01),
+                    );
+                    ui.end_row();
+                    ui.checkbox(&mut new_circle_collision.view, "View".to_string());
+                    ui.end_row();
+                });
+        });
+
+        if let Some(new_map) = serde_json::to_value(new_circle_collision)
+            .ok()
+            .map(|v| v.as_object().cloned())
+            .flatten()
+        {
+            for (k, v) in new_map {
+                self.config.as_object_mut().unwrap().insert(k, v);
+            }
+        }
+    }
+}
+
+fn update_view_circle_collision(
+    model_instances: Query<&Affiliation<Entity>, (With<ModelMarker>, Without<Group>, With<Robot>)>,
+    model_descriptions: Query<&ModelProperty<Robot>, (With<ModelMarker>, With<Group>)>,
+    selection: Res<Selection>,
+    poses: Query<&Pose>,
+    mut gizmos: Gizmos,
+) {
+    let Some(selected_entity) = selection.0 else {
+        return;
+    };
+    let mut description_entity: Option<Entity> = None;
+    if model_descriptions.get(selected_entity).ok().is_some() {
+        description_entity = Some(selected_entity);
+    } else {
+        if let Some(affiliation) = model_instances.get(selected_entity).ok().and_then(|a| a.0) {
+            if model_descriptions.get(affiliation).is_ok() {
+                description_entity = Some(affiliation);
+            }
+        }
+    }
+    let Some(ModelProperty(robot)) =
+        description_entity.and_then(|e| model_descriptions.get(e).ok())
+    else {
+        return;
+    };
+
+    let Some(circle_collision) = robot
+        .properties
+        .get(&Collision::label())
+        .and_then(|c| serde_json::from_value::<Collision>(c.clone()).ok())
+        .filter(|c| c.kind() == CircleCollision::label())
+        .and_then(|c| serde_json::from_value::<CircleCollision>(c.config().clone()).ok())
```

Comment:
Since egui is an [immediate mode](https://en.wikipedia.org/wiki/Immediate_mode_(computer_graphics)) GUI framework, we're going to be parsing these values on every frame update. That's probably not an enormous burden given how lightweight these structs are, but I worry about the scalability in general.

We can move along with this PR as-is, but we should think about how to find the right balance with this extensible structure. For example, maybe `CircleCollision` should be a component which we add or remove from the entity as needed based on changes in the properties inside `Robot`. Then this widget can query for `CircleCollision` instead of doing this parsing.

Reply:
That's a good point. I'll follow up on this.